File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
from collections import defaultdict
def solution(nums):
    answer = 0
    pocketDict = defaultdict(int)
    for num in nums:
        pocketDict[num] += 1
        
    n = len(nums) // 2
    return n if len(pocketDict.keys()) >= n else len(pocketDict.keys())

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int choiceNum = nums.length / 2;
        Set<Integer> numSet = new HashSet<>();
        for(int num : nums) numSet.add(num);
        
        return numSet.size() <= choiceNum ? numSet.size() : choiceNum;
    }
}